tests/test_group_api.py

```diff
@@ -215,6 +215,96 @@ def test_get_group_users(self):
         users = [User.provision_user(f"user_{i}", groups=[name]).name for i in range(10)]
         self._test_paging(f'/v1/group/{name}/users', headers, 5, key)
 
+    def test_put_users(self):
+        users = [User.provision_user(f"user_{i}").name for i in range(11)]
+        tests = [
+            {
+                'action': 'add',
+                'json_request_body': {
+                    "users": [users[0]]
+                },
+                'responses': [
+                    {'code': 200},
+                    {'code': 304}
+                ]
+            },
+            {
+                'action': 'remove',
+                'json_request_body': {
+                    "users": [users[0]]
+                },
+                'responses': [
+                    {'code': 200},
+                    {'code': 304}
+                ]
+            },
+            {
+                'action': 'add',
+                'json_request_body': {
+                    "users": users[:-1]
+                },
+                'responses': [
+                    {'code': 200},
+                    {'code': 304}
+                ]
+            },
+            {
+                'action': 'remove',
+                'json_request_body': {
+                    "users": users[:-1]
+                },
+                'responses': [
+                    {'code': 200},
+                    {'code': 304}
+                ]
+            },
+            {
+                'action': 'add',
+                'json_request_body': {
+                    "users": users
+                },
+                'responses': [
+                    {'code': 400},
+                    {'code': 400}
+                ]
+            },
+            {
+                'action': 'remove',
+                'json_request_body': {
+                    "users": users
+                },
+                'responses': [
+                    {'code': 400},
+                    {'code': 400}
+                ]
+            }
+        ]
+        for i in range(len(tests)):
```

Comment:
Considering line 283, I think it makes more sense to write

```suggestion
        for i, test in enumerate(tests):
```

Reply:
that's a new one for me. Thank you. I'll be using that in the future.